src/layers/GeomodelLayer.ts

```diff
@@ -0,0 +1,146 @@
+import curveCatmullRom from 'cat-rom-spline';
+import { ScaleLinear } from 'd3-scale';
+import { Graphics } from 'pixi.js';
+import WebGLLayer from './WebGLLayer';
+import {
+  GeoModelData,
+  GeomodelLayerOptions,
+  OnUpdateEvent,
+  OnRescaleEvent,
+} from '../interfaces';
+
+class GeomodelLayer extends WebGLLayer {
+  options: GeomodelLayerOptions;
+
+  curveOptions = { samples: 50, knot: 0.5 };
+
+  constructor(id: string, options: GeomodelLayerOptions) {
+    super(id, options);
+    this.options = {
+      ...options,
+    };
+    this.render = this.render.bind(this);
+  }
+
+  onUpdate(event: OnUpdateEvent) {
+    super.onUpdate(event);
+    this.render(event);
+  }
+
+  onRescale(event: OnRescaleEvent) {
+    super.onRescale(event);
+    this.render(event);
+  }
+
+  render(event: OnRescaleEvent | OnUpdateEvent) {
+    const { xscale, yscale, data } = event;
+    const scaledData = this.generateScaledData(data, xscale, yscale);
+
```

Comment:
Is it possible to set the transformation in pixi? If so, setting the transformation to scale and translation would make it possible to draw in world coordinates and avoid having to scale each coordinate, making it easier to read and maintain the code.